### PR TITLE
Oric1 Software List

### PR DESCRIPTION
**Pull request by enewton**
I'm trying to put together a software list for the Oric1.

I did ask on the forum some questions but not sure how much traffic that gets: https://forum.mamedev.org/viewtopic.php?f=14&t=894&sid=665161912258e79bd1d640c551908ed5

I've pressed ahead, assuming this doesn't already exist and is useful.

My main problem: I can't get Mame to recognise the file. Do I need to add a reference to it somewhere else in the source, or name it subtly different, or something wrong in the header of the file? E.g. oric vs oric1

Can anyone help please?

### CODE REVIEW

**Comment by galibert** [comment]
You need that patch:
[oric.txt](https://github.com/mamedev/mame/files/10974668/oric.txt)

**Comment by simzy39** [comment]
@enewton If you can, join https://forums.bannister.org/ubbthreads.php?ubb=newuser

**Comment by enewton** [comment]
Thanks @galibert that worked. I'll add it to this PR.

Thanks @simzy39 . I've registered, but it may need manual approval as it won't let me log in. Is that a better 'dev' forum than forum.mamedev.org?

**Review by cuavas — COMMENTED** [review]
From a quick look:
* This won’t pass validation with blank descriptions/years/publishers
* Use one tab per indentation level, and indent your `<software>` elements as they aren‘t at root level
* Remember to add the CC0 license header as in the other software lists
* “Oric 1” has a space in it

**Comment by cuavas** [comment]
Also, you can mark drafts as drafts.

**Comment by enewton** [comment]
> Also, you can mark drafts as drafts.

Yep, sorry, I was looking for the button for that but couldn't see it.

File: hash/oric1_cass.xml [hunk 1]
```diff
@@ -0,0 +1,195 @@
+<?xml version="1.0"?>
+<!DOCTYPE softwarelist SYSTEM "softwarelist.dtd">
+<!--
+license:CC0-1.0
+-->
+<softwarelist name="oric1" description="Oric 1 cassettes">
```
Comment (cuavas) [inline hash/oric1_cass.xml:6]: The short name should match the file name – I think it fails a validity check if it doesn’t.

File: hash/oric1_cass.xml [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+	<software name="xenon1a2" cloneof="xenon1" supported="no">
+		<description>Xenon 1 [a2]</description>
+		<year>1983</year>
+		<publisher>IJK Software</publisher>
+		<part name="cass" interface="oric1_cass">
+			<dataarea name="cass" size="37396">
+				<rom name="Xenon 1 (UK) (1983) [a2].tap" size="37396" crc="e8536524" sha1="de6dc28770b8e900a427ccdf5c6d315a2ca93c0b"/>
+			</dataarea>
+		</part>
+	</software>
```
Comment (cuavas) [inline hash/oric1_cass.xml:39]: Can you please not copy markup from the file names into descriptions?  Please follow the conventions in other software lists for disambiguation text.

File: src/mame/tangerine/oric.cpp [hunk 3]
```diff
@@ -37,6 +37,7 @@
 #include "emupal.h"
 #include "screen.h"
 #include "speaker.h"
+#include "softlist_dev.h"
```
Comment (cuavas) [inline src/mame/tangerine/oric.cpp:40]: Please put the headers from src/emu in alphabetical order here.  It’s easier to keep track of them that way.

File: hash/oric1_cass.xml (hunk 1 above)
Comment (enewton) [reply hash/oric1_cass.xml]: Done

File: hash/oric1_cass.xml (hunk 2 above)
Comment (enewton) [reply hash/oric1_cass.xml]: Done. (I think)
Not sure what software name to use for say Xenon 1 (Alt 1)

File: src/mame/tangerine/oric.cpp (hunk 3 above)
Comment (enewton) [reply src/mame/tangerine/oric.cpp]: Done

**Comment by ArcadeShadow** [comment]
Please, change the descriptive text, on the software description, like (Alt), to lowercase (alt)

**Comment by galibert** [comment]
Mame is never supposed to segfault.  Which one for instance?

  OG.

**Comment by enewton** [comment]
Mushroom mania is quite a good one as it’s fairly quick to load.


On 22 Mar 2023, at 09:36, Olivier Galibert ***@***.***> wrote:

﻿

Mame is never supposed to segfault. Which one for instance?

OG.

**Comment by galibert** [comment]
Ok, I've found an image, and mame is not segfaulting but the cpu stops because it encounters an 0xf2 opcode which kills a 6502 (that opcode never ends).

**Comment by cuavas** [comment]
> Question: How much testing do I/we need to do to mark broken games as 'not working'?

In general, games should be marked not working if it isn’t possible to complete a game as intended by the designers/developers (taking original game bugs into consideration).  Not all the software in the software lists is tested exhaustively – just do your best.

I patched up some of the metadata because I want to get this in for the release:
* Put articles at the beginning of titles where they belong.
* Put names in reading order.
* Use uppercase for proper nouns in disambiguation text.
* Make short names for clones by appending a suffix to the parent’s short name.
* Use multi-part items for multiple software that came on multiple tapes.

**Comment by angelosa** [comment]
> Question: How much testing do I/we need to do to mark broken games as 'not working'?

The basic QA gist:
- `supported="no"` for anything that cannot be mounted, outright crashes, has unusable fundamental parts (cfr. option mode for console games) or has breaking bugs that will _ruin the immersion_ (cfr. a quiz game where you can't see the text);
- `supported="partial"` for stuff that has minor GFX/sound glitches, has unsupported optional card slots or cannot be saved (provided they are humanly possible to clear them, I won't give a `partial` for an RPG for instance). Special mention for multigame collections: if the vast majority works and only a few exhibit stuff that would otherwise mark them with a `no` if stand-alone then it's fine to mark them `partial` instead.
- `supported="yes"` is where all of the above doesn't happen, can be completed etc. It's otherwise fine to just not define the flag assuming above isn't encountered, given that obviously there are places where domain and/or language knowledge is a must.
- `no` and `partial` always require a reason, either as a `notes` bucket or as XML comment near the node in question. I tend to put each issue encountered on separate lines, as if they are bullet lists, ordered by severity levels. I prefer to put those in notes rather than XML comments because latter are tricky to parse externally.

**Comment by enewton** [comment]
Yep sorry, I've had a busy week so not had much time to work on it.
I'll test some more games and raise diffs to add any not-working tags.

**Comment by cuavas** [comment]
@enewton is the Cross Bomber tape available somewhere?  The tape image in the software list doesn’t seem to match the image in the release on Fabrizio’s GitHub project.

**Comment by Tafoid** [comment]
@enewton 
There is one image that was not able to be obtained - can you provide it?

Cross Bomber [folder: crossbomber - size: 6kb]
missing set: Cross Bomber
missing rom: Cross Bomber (2021-05-13)(Caruso, Fabrizio)[48K].tap [size: 6412] [CRC32: 5915840a] [SHA1: ab594aa3f6913c836956854374eafe243e4b457c]

The one on the author's GitHub page is dated later (Jul 20, 2022) and the matching date in the entry above could not be found.
Is this a mistake or is there another reason?

Please send to the email provided [HERE](https://wiki.mamedev.org/index.php/Submitting_Source_Code)

Thank you

**Comment by enewton** [comment]
Hi,
I never had an image. The bulk of the CRCs came from the spludlow database, so if you’ve found a better image use that.

Cheers,
Erik

On 27 Mar 2023, at 17:40, Scott Stone ***@***.***> wrote:

﻿

@enewton<https://github.com/enewton>
There is one image that was not able to be obtained - can you provide it?

Cross Bomber [folder: crossbomber - size: 6kb]
missing set: Cross Bomber
missing rom: Cross Bomber (2021-05-13)(Caruso, Fabrizio)[48K].tap [size: 6412] [CRC32: 5915840a] [SHA1: ab594aa3f6913c836956854374eafe243e4b457c]

The one on the author's GitHub page is dated later (Jul 20, 2022) and the matching date in the entry above could not be found.
Is this a mistake or is there another reason?

Please send to the email provided HERE<https://wiki.mamedev.org/index.php/Submitting_Source_Code>

Thank you

**Comment by cuavas** [comment]
> I never had an image. The bulk of the CRCs came from the spludlow database, so if you’ve found a better image use that.

You can’t add software list items for things that don’t exist.

Apparently Cross Bomber was updated after the original release, with the MC1000 version being updated a second time.  It would be nice to actually get a hold of the original release as it seems to have disappeared entirely from the internet.

**Comment by enewton** [comment]
I've found 2 versions on the game author's GitHub page but neither match the CRC. I think we should just update the XML to match the latest version.

**Comment by cuavas** [comment]
> I've found 2 versions on the game author's GitHub page but neither match the CRC. I think we should just update the XML to match the latest version.

Yes, because he replaced it with a rebuilt binary a year after it was released.  It would still be nice to dig up the original release of that version.

**Comment by DavidHaywood** [comment]
For what it's worth, I think this could do with a cleanup pass marking cracked games as such.  The Manic Miner version for example is cracked, rather than an original tape and the copy protection can be passed just by pressing enter.

There's nothing wrong with such versions, as they were likely more widespread than the original software, but they should be marked as such so that we know the original tapes are still needed.

Unfortunately a lot of 'less popular' systems have this issue, where the only dumps around are cracked versions, the situation with something with the Sam Coupe is just as bad and many of those aren't properly marked as cracked versions either.

**Comment by DavidHaywood** [comment]
also maybe a QA pass as 'ghostman' just crashes MAME after loading (not just freezes, crashes MAME after starting to show game text, although I don't have a backtrace right now)

software that doesn't work should be marked as unsupported

**Comment by smf-** [comment]
On 27/03/2023 17:40, Scott Stone wrote:
> Cross Bomber [folder: crossbomber - size: 6kb]
>
> missing set: Cross Bomber
> missing rom: Cross Bomber (2021-05-13)(Caruso, Fabrizio)[48K].tap 
> [size: 6412] [CRC32: 5915840a] [SHA1: 
> ab594aa3f6913c836956854374eafe243e4b457c]
>
> The one on the author's GitHub page is dated later (Jul 20, 2022) and 
> the matching date in the entry above could not be found.
> Is this a mistake or is there another reason?
>
Could be this...

https://github.com/Fabrizio-Caruso/CROSS-LIB/releases/tag/XBomber

And Message ID: ***@***.***> 
Fabrizio-Caruso <https://github.com/Fabrizio-Caruso>released thisMay 13, 
2021

which seems to be an updated version of

https://github.com/Fabrizio-Caruso/CROSS-LIB/releases/tag/bomber

Fabrizio-Caruso <https://github.com/Fabrizio-Caruso>released thisAug 26, 
2020

Though Xbomber_atmos_and_oric1_48k.tap 
<https://github.com/Fabrizio-Caruso/CROSS-LIB/releases/download/XBomber/Xbomber_atmos_and_oric1_48k.tap> 
in either of those matches the crc

I wonder if there is some kind of tap clean that the original was put 
through?

**Comment by Tafoid** [comment]
This particular missing image has been found in a TOSEC list on archive.org - just not re-added as of yet.  If you'd like to do this, contact me privately otherwise it lays on my pile of stuff to do eventually.

**Pull request merged**